### PR TITLE
Removes depends_on tag from docker-compose.yml files and documentation

### LINKED ISSUE #198: docker-compose.yml uses depends_on (which is ignored by stack deploy)
In the version _47d0fdcf_ of the file [faas/docker-compose.yml](https://github.com/alexellis/faas/blob/47d0fdcf6cc23612f246094c193fd8a1de7fd94d/docker-compose.yml) in several services `depends_on` is used. 
In [deploy_stack.sh](https://github.com/alexellis/faas/blob/47d0fdcf6cc23612f246094c193fd8a1de7fd94d/deploy_stack.sh) this file used with `docker stack deploy`.  
Unfortunately the compose file reference points out that `depends_on` is ignored then deploying a stack using a compose file version 3.x, see https://docs.docker.com/compose/compose-file/#depends_on

I've not tested it but possibly this can result in some problems setting up faas on Swarm.

### PR DESCRIPTION
**Pull request by VincenzoFerme**
## Description
Removes depends_on tag from docker-compose.yml files and documentation

## Motivation and Context
Fixes #198, and updated `DEV.md` documentation accordingly.

## How Has This Been Tested?
Executing `docker stack --compose-file=docker-compose.yml deploy` for all the changed files and verifying all the services were up and runnin. 

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by alexellis** [comment]
Looks like you checked the box " I have signed-off my commits with git commit -s" but haven't signed-off the commits.

**Comment by VincenzoFerme** [comment]
@alexellis done, sorry, I did verify it but I did not sign it off.

**Comment by alexellis** [comment]
LGTM

File: docker-compose.arm64.yml [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (johnmccabe) [inline docker-compose.arm64.yml]: Unneeded blank line here

File: docker-compose.extended.armhf.yml [hunk 2]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (johnmccabe) [inline docker-compose.extended.armhf.yml]: Unneeded blank line

**Comment by VincenzoFerme** [comment]
@johnmccabe I updated it

**Comment by alexellis** [comment]
Merged

**Pull request merged**